lib/roo_on_rails/rack/populate_env_from_jwt.rb

```diff
@@ -0,0 +1,98 @@
+require 'json/jwt'
+require 'faraday'
+require 'faraday_middleware'
+
+module RooOnRails
+  module Rack
+    class PopulateEnvFromJWT
+      UnnacceptableKeyError = Class.new(RuntimeError)
+      VALID_JWK_URL_PREFIXES = [
+        'https://www.deliveroo.com/identity-keys/',
+        'https://identity.deliveroo.com/identity-keys/'
+      ].freeze
+
+      def initialize(app, logger:)
+        @app = app
+        @keys = {}
+        @logger = logger
+
+        if development?
+          @logger.warn "JWTs will not have their signatures checked, you're in development mode."
+        end
+      end
+
+      def call(env)
+        env['roo.identity'] = decode_authorization_header(env['HTTP_AUTHORIZATION'])
+        @app.call(env)
+
+      # Other exceptions will bubble up, allowing the higher middleware to return a 500, which is
+      # intentional.
+      rescue UnnacceptableKeyError, JSON::JWT::Exception => e
+        # Identifying user is clearly attempting to hack or has been given a totally incorrect
+        # token, log this and flag as Forbidden, without executing the rest of the middleware stack.
+        ::NewRelic::Agent.notice_error(e) if defined?(NewRelic)
+        [403, {}, []]
+      end
+
+      private
+
+      def development?
+        if ENV['RACK_ENV'].nil?
+          @logger.warn "Your RACK_ENV isn't set. You probably want it set to 'development' in dev."
+        end
+
+        ENV['RACK_ENV'] == 'development'
+      end
+
+      # @raise [UnnacceptableKeyError,Faraday::Error,OpenSSL::OpenSSLError] From `#public_key`
+      # @raise [JSON::JWT::Exception] Bubble ups from `JSON::JWT.decode`
+      # @return [JSON::JWT] The list of claims this header makes by way of a JWS token. Will be an
+      #   empty hash for invalid or absent tokens.
+      def decode_authorization_header(header_value)
+        return JSON::JWT.new unless (header_value || '').starts_with?('Bearer ')
```

Comment:
Can you use `header_value&.starts_with?('Bearer ')` here, or are we using too old a version of Ruby for RoR?

Reply:
Roo On Rails supports old rubies (for now) :(